applications/FluidDynamicsApplication/tests/cpp_tests/test_two_fluid_navier_stokes_element_volume_error_source.cpp

```diff
@@ -0,0 +1,1836 @@
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez, Ruben Zorrilla
+//
+//
+
+// System includes
+#include <set>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "containers/model.h"
+#include "spaces/ublas_space.h"
+#include "includes/properties.h"
+#include "includes/model_part.h"
+#include "utilities/math_utils.h"
+#include "includes/global_pointer_variables.h"
+#include "custom_elements/two_fluid_navier_stokes.h"
+#include "custom_constitutive/newtonian_2d_law.h"
+#include "custom_constitutive/newtonian_3d_law.h"
+#include "custom_constitutive/newtonian_two_fluid_3d_law.h"
+#include "fluid_dynamics_application_variables.h"
+#include "processes/find_nodal_neighbours_process.h"
+#include "utilities/normal_calculation_utils.h"
+
+namespace Kratos {
+	namespace Testing {
+
+		typedef ModelPart::IndexType									 IndexType;
+		typedef ModelPart::NodeIterator					          NodeIteratorType;
+
+	    /** Checks the TwoFluidNavierStokes2D3N element.
+	     * Checks the LHS and RHS computation
+	     */
+	    KRATOS_TEST_CASE_IN_SUITE(ElementTwoFluidNavierStokes2D3NError, FluidDynamicsApplicationFastSuite)
+		{
+
+			Model current_model;
+			ModelPart& modelPart = current_model.CreateModelPart("Main");
+			modelPart.SetBufferSize(3);
+
+			// Variables addition
+			modelPart.AddNodalSolutionStepVariable(BODY_FORCE);
+			modelPart.AddNodalSolutionStepVariable(DENSITY);
+			modelPart.AddNodalSolutionStepVariable(DYNAMIC_VISCOSITY);
+			modelPart.AddNodalSolutionStepVariable(DYNAMIC_TAU);
+			modelPart.AddNodalSolutionStepVariable(PRESSURE);
+			modelPart.AddNodalSolutionStepVariable(VELOCITY);
+			modelPart.AddNodalSolutionStepVariable(MESH_VELOCITY);
+			modelPart.AddNodalSolutionStepVariable(DISTANCE);
+
+			// Process info creation
+			double delta_time = 0.1;
+			modelPart.GetProcessInfo().SetValue(DYNAMIC_TAU, 0.001);
+			modelPart.GetProcessInfo().SetValue(SOUND_VELOCITY, 1.0e+3);
+			modelPart.GetProcessInfo().SetValue(DELTA_TIME, delta_time);
+			modelPart.GetProcessInfo().SetValue(VOLUME_ERROR,10.0);
+			
+			Vector bdf_coefs(3);
+			bdf_coefs[0] = 3.0/(2.0*delta_time);
+			bdf_coefs[1] = -2.0/delta_time;
+			bdf_coefs[2] = 0.5*delta_time;
+			modelPart.GetProcessInfo().SetValue(BDF_COEFFICIENTS, bdf_coefs);
+
+			// Set the element properties
+			Properties::Pointer pElemProp = modelPart.CreateNewProperties(0);
+			pElemProp->SetValue(DENSITY, 1000.0);
+			pElemProp->SetValue(DYNAMIC_VISCOSITY, 1.0e-05);
+			Newtonian2DLaw::Pointer pConsLaw(new Newtonian2DLaw());
+			pElemProp->SetValue(CONSTITUTIVE_LAW, pConsLaw);
+
+			// Geometry creation
+			modelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+			modelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+			modelPart.CreateNewNode(3, 0.0, 1.0, 0.0);
+			std::vector<ModelPart::IndexType> elemNodes {1, 2, 3};
+			modelPart.CreateNewElement("TwoFluidNavierStokes2D3N", 1, elemNodes, pElemProp);

```

Comment:
You can encapsulate this within an auxiliary function to avoid repeating it in all the tests.

Reply:
Removed this test and added to existing file. Done!